### PR TITLE
Adding scoring capability in image captioning module

### PR DESCRIPTION
**Pull request by lorisbaz**
Extend scoring features originally implemented for MT to the image captioning module. Now given an image and a sentence, one can obtain a score. This feature is useful to perform image-to-text and text-to-image retrieval operations.

#### Pull Request Checklist ##
- [x] Changes are complete (if posting work-in-progress code, prefix your pull request title with '[WIP]'
until you can check this box.
- [x] Unit tests pass (`pytest`)
- [x] ~~Were system tests modified? If so did you run these at least 5 times to account for the variation across runs?~~
- [x] System tests pass (`pytest test/system`)
- [x] Passed code style checking (`./style-check.sh`)
- [x] You have considered writing a test
- [x] Updated major/minor version in `sockeye/__init__.py`. Major version bump if this is a backwards incompatible change.
- [x] Updated CHANGELOG.md

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by lorisbaz** [comment]
@tdomhan @fhieber @davvil can anybody have a look? It should not be a big review, since it extends the existing scoring capabilities.

File: sockeye/image_captioning/scoring.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (tdomhan) [inline sockeye/image_captioning/scoring.py]: 2019 ;)

File: sockeye/image_captioning/score.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (tdomhan) [inline sockeye/image_captioning/score.py]: 2019

File: sockeye/image_captioning/score.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+from .. import model
+from .. import scoring
+from . import scoring as scoring_images
+from .. import utils
+from .. import vocab
+from ..log import setup_main_logger
+from ..output_handler import get_output_handler
+from ..utils import check_condition
+from .train import read_feature_shape
+from.captioner import _extract_features
```
Comment (tdomhan) [inline sockeye/image_captioning/score.py]: spacing seems a bit off (`from.captioner`)

File: sockeye/image_captioning/scoring.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+from .. import vocab
+from ..inference import TranslatorInput, TranslatorOutput
+from ..output_handler import OutputHandler
+
+logger = logging.getLogger(__name__)
+
+
+class Scorer:
+    """
+    Scorer class takes a ScoringModel and uses it to score a stream of parallel sentences.
```
Comment (tdomhan) [inline sockeye/image_captioning/scoring.py]: "parallel sentences" -> "image + sentence pair"

File: sockeye/image_captioning/scoring.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    :param model: The model to score with.
+    :param source_vocabs: The source vocabularies.
+    :param target_vocab: The target vocabulary.
+    """
+    def __init__(self,
+                 model: ScoringModel,
+                 source_vocabs: List[vocab.Vocab],
+                 target_vocab: vocab.Vocab,
+                 constant_length_ratio: float = -1.0) -> None:
+        #self.source_vocab_inv = vocab.reverse_vocab(source_vocabs[0])
```
Comment (tdomhan) [inline sockeye/image_captioning/scoring.py]: could be removed

File: sockeye/image_captioning/scoring.py (hunk 1 above)
Comment (lorisbaz) [reply sockeye/image_captioning/scoring.py]: Thanks :)

**Comment by lorisbaz** [comment]
Covered @tdomhan comments. It should be ready to be merged

**Comment by fhieber** [comment]
Thanks for the contribution!

**Comment by tdomhan** [comment]
Yeah, I'm not sure why it's failing. We have been seeing some timeouts from Travis unfortunately.

**Pull request merged**